scripts/lib.sh

```diff
@@ -41,6 +41,7 @@ get_from_kata_deps() {
 	# through CI. For the kernel, .ci/install_kata_kernel.sh file in tests
 	# repository will pass the kernel version as an override to this function to
 	# allow testing of kernels before they land in tree.
+  CI=${CI:-false}
```

Comment:
Having this fix here feels odd. I think it is having the `${x:-y}` idiom in the middle of a func acting on a global var that is making my senses twitch...  I had a surf, and learnt some new bash - how about:

```
if [ ! -z ${CI+x} ]; then ...
```

I'm guessing if use of `${x:-y}` does not trip on the `set -u`, then use of `${x+y}` should not either.

Reply:
How ever which we I try not to create the local variable, the script always fails with the unbound variable error. It does seem like we need to create all the variables locally before we do anything. I can move the definition up, but i guess that will apply to all the variables we create in this file.